assets/js/study.js

```diff
@@ -183,23 +218,156 @@ function give_hints(once) {
     return false;
 }
 
+/**
+ * Create a playable arrow.
+ * 
+ * @param  m  The move to create an arrow for.
+ * @param  max  The number of times the most played candidate move has been played.
+ * @param  min  The number of times the least played candidate move has been played.
+ * @param  pgn_color  The color of the arrow to be created, or undefined for classic playable only arrows.
+ */
+function create_playable_arrow(m, max, min, pgn_color = undefined) {
+    let move = m.move;
+    let some_neglected = some_moves_neglected(max, min);
+    let current_neglected = current_move_neglected(m, max);
+    let brush = undefined;
+    if(pgn_color) {  // pgn arrows
+        let brush_prefix = some_neglected ? (current_neglected ? "playable_pgn_normal_" : "playable_pgn_transparent_") : "playable_pgn_normal_";
+        brush = brush_prefix + pgn_color;
+    } else {  // auto arrows
+        brush = some_neglected ? (current_neglected ? "normal" : "transparent") : "normal";
+    }
+    return create_arrow_from_move(move, brush);
+}
+
+/**
+ * Combines all playable moves and all PGN arrows from the PGN file and returns the intersection of
+ * those two sets, ie all PGN arrows that are also playable moves. A special object containing information
+ * from both the move and the PGN arrow is returned.
+ */
+function get_doubled_playable_move_objects(all_moves, all_pgn_arrows) {
+    let all_pgn_arrow_uci = [];
+    let color_by_pgn_arrow_uci = {};
+    for (let cal of all_pgn_arrows) {
+        let ftc = cal_to_ftc(cal);
+        let ucistr = ftc.from + ftc.to;
+       
+        color_by_pgn_arrow_uci[ucistr] = ftc.color;
+        all_pgn_arrow_uci.push(ucistr);
+    }
+
+    // Merge the playable moves and the pgn arrows
+    let union = all_moves.map(move => {
+        let uci = move_to_ucistr(move);
+        let color = color_by_pgn_arrow_uci[uci];
+        return {
+            color: color,
+            uci: uci,
+            move: move
+        };
+    })
+    // And filter so that only those moves that are both playable a pgn arrow are left
+    return union.filter(x => array_contains(all_pgn_arrow_uci, x.uci))
+}
+
+/**
+ * Add overlays to explain when certain arrow combinations appear.
+ */
+function setup_arrow_overlays(all_moves, clean_pgn_arrows, max, min, only_pgn = false) {
```

Comment:
I wonder if this logic should get moved to a specialized subclass of TextOverlayManager. I personally would prefer it to more neatly separate the concerns of the study from the Hinting logic.

Reply:
Good point! 

I broke out the arrow hint code and put it all in a new file called overlays.js and moved some of the functions from study.js into the TextOverlayManager class. study.js is a bit cleaner now.

Previously I had mixed the storage of seen suggestions with the storage of seen overlays because it was convenient to re-use some of the suggestion code, but I've now split these into two different things because they really are two different things. So the suggestions work just like before (but I did keep some of the code changes for them to avoid collisions between multiple suggestions). And then I added functions to TextOverlayManager to take care of remembering the current overlays (so that they can be re-added if the board is cleared). Becuase overlays are always only showed one time there was really no need to keep a separate session memory of which overlays had been shown. It's all in localstorage now.

Note! For the reason mentioned above, to test this you now how to clear the localstorage keys "[STUDYID]_overlays_[OVERLAYID)". Just search for overlays and you will find them. And because of that I've now changed this PR from a draft to a real PR, so you can merge whenver you feel like.

I also moved some arrow drawing functions into ground.js where I think they belong better, so study.js is much cleaner now.